Content.Shared/SubFloor/SharedTrayScannerSystem.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.SubFloor;
+
+public abstract class SharedTrayScannerSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TrayScannerComponent, ComponentGetState>(OnTrayScannerGetState);
+        SubscribeLocalEvent<TrayScannerComponent, ComponentHandleState>(OnTrayScannerHandleState);
+    }
+
+    public void ToggleTrayScanner(EntityUid uid, bool state, TrayScannerComponent? scanner = null)
+    {
+        if (!Resolve(uid, ref scanner))
+            return;
+
+        scanner.Toggled = state;
+        scanner.Dirty();
+
+        RaiseLocalEvent(uid, new TrayScannerToggleEvent(scanner.Toggled));
```

Comment:
As far as I can tell this event just exists to call a single method on the client-side system. Unless there's some reason it needs to be an event, you should probably remove it and just have the client override `ToggleTrayScanner` and move the `OnTrayScannerToggle` logic there.